verify.go

```diff
@@ -0,0 +1,114 @@
+// This program loads a group of certificates provided on the command line
+// and verifies certain properties for consistency between certificates:
+// - All certificates with the same Issuer string should have the same RawIssuer
+// - All certificates with the same Subject string should have the same RawIssuer
+// - All certificates with the Issuer should have the same CRL and AIA Issuer URLs
+// - All certificates iwth the same Subject should have the same public key.
+package main
+
+import (
+	"bytes"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+)
+
+// loadCert loads a PEM certificate specified by filename or returns an error
+func loadCert(filename string) (cert *x509.Certificate, err error) {
+	certPEM, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return
+	}
+	block, _ := pem.Decode(certPEM)
+	if block == nil {
+		return nil, fmt.Errorf("No data in cert PEM file %s", filename)
+	}
+	cert, err = x509.ParseCertificate(block.Bytes)
+	return
+}
+
+func main() {
+	if err := main2(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+type myCert struct {
+	*x509.Certificate
+	filename string
+}
+
+func main2() error {
+	var certs []myCert
+	for _, fn := range os.Args[1:] {
+		cert, err := loadCert(fn)
+		if err != nil {
+			return err
+		}
+		certs = append(certs, myCert{cert, fn})
+	}
+
+	var certsByIssuer = make(map[string][]myCert)
+	var certsBySubject = make(map[string][]myCert)
+	for _, c := range certs {
+		certsByIssuer[c.Issuer.String()] = append(certsByIssuer[c.Issuer.String()], c)
+		certsBySubject[c.Subject.String()] = append(certsBySubject[c.Subject.String()], c)
+	}
+	fmt.Println("\nCerts by issuer:")
+	for k, v := range certsByIssuer {
+		fmt.Println(k)
+		var rawIssuer = v[0].RawIssuer
+		var issuerURL, crlURL string
+		for _, c := range v {
+			fmt.Println("  ", c.Subject)
+			if !bytes.Equal(rawIssuer, c.RawIssuer) {
+				return fmt.Errorf("%s has a mismatch in raw issuer", c.filename)
+			}
+			if len(c.CRLDistributionPoints) > 0 {
```

Comment:
This should be checking v[0].CRLDistributionPoints (or even better, any(c.CRLDistributionPoints for c in v), so that this check doesn't get turned on/off by each individual cert. Same with similar checks below.